### ISSUE TITLE
Variable timezones not parsing correctly.

### ISSUE THREAD

**Issue by ceeeekay** [logstash-plugins/logstash-filter-date#58]
We have syslog logs shipping from multiple timezones, so need to use the timezone directive in the date filter.

In order to keep configs sane I've tried using `timezone => %{timezone}` but the filter doesn't parse the date correctly.

Basic config:
>$ cat stdin-timezone.conf
input {
  stdin {
    codec => json
  }
}
filter {
   date {
    match => [
      "timestamp",
        "MMM  d HH:mm:ss",
        "MMM dd HH:mm:ss"
    ]
    timezone => "%{timezone}"
  }
}
output {
  stdout {
    codec => rubydebug {}
  }
}

Pasting sample lines containing the same time and two different timezones gives incorrect results:
>$ /opt/logstash/bin/logstash -f stdin-timezone.conf 
Settings: Default filter workers: 4
Logstash startup completed

input: `{"timezone": "Pacific/Auckland","timestamp": "Mar  1 12:25:45"}`
>{
      "timezone" => "Pacific/Auckland",
     "timestamp" => "Mar  1 12:25:45",
      "@version" => "1",
    "@timestamp" => "2000-02-29T23:25:45.000Z",
          "host" => "civet"
}

input: `{"timezone": "Asia/Hong_Kong","timestamp": "Mar  1 12:25:45"}`
>{
      "timezone" => "Asia/Hong_Kong",
     "timestamp" => "Mar  1 12:25:45",
      "@version" => "1",
    "@timestamp" => "2000-03-01T04:25:45.000Z",
          "host" => "civet"
}

Specifying a timezone directly in the config, e.g., `timezone => "Asia/Hong_Kong"` gives the expected result:
input: `{"timezone": "Asia/Hong_Kong","timestamp": "Mar  1 12:25:45"}`
>{
      "timezone" => "Asia/Hong_Kong",
     "timestamp" => "Mar  1 12:25:45",
      "@version" => "1",
    "@timestamp" => "2016-03-01T04:25:45.000Z",
          "host" => "civet"
}

Versions:
Logstash 2.2.2
Date filter 2.1.2

**Comment by ceeeekay** [comment]
@torrancew, FYI.

**Comment by wiibaa** [comment]
Can you please check the locale configuration of your system. I suspect you are hitting an issue with the english fallback parser. 
If so, please add `locale => "en" ` to your date filter configuration

**Comment by ceeeekay** [comment]
Hi @wiibaa I've added `locale => "en"` but still get the same year 2000 date results.

**Comment by wiibaa** [comment]
Hi @ceeeekay , you spot a bug! 
I submitted a fix, if you need a workaround to inject the current year you could do something like

````
mutate {
    replace => { "timestamp" => "%{+yyyy} %{timestamp}" }
  }
````
and add `yyyy` to your patterns in the date filter

**Comment by ceeeekay** [comment]
@wiibaa, awesome - thank you for getting it sorted.
I'll wait for the patch to reach GA, as I don't want to edit all my syslog filters if there is a fix on the way.

Thanks again!

**Comment by wiibaa** [comment]
published in 3.0.1

@suyograo can you close please
